bin/upgrade/24.05/scripts/generate_crashgroup_app_version_list.js

```diff
@@ -3,94 +3,62 @@
                         },
                     },
-                },
-            });
-
-            return acc;
-        }, []);
-    }
-
-    console.log('Applying updates');
-
-    const BATCH_SIZE = 200;
-    let errCount = 0;
-
-    for (let idx = 0; idx < apps.length; idx += 1) {
-        const appId = apps[idx]._id;
-        const crashgroupCollection = `app_crashgroups${appId}`;
-
-        if (updates[appId]) {
-            let buffer = [];
-
-            for (let idy = 0; idy < updates[appId].length; idy += 1) {
-                buffer.push(updates[appId][idy]);
-
-                if (buffer.length === BATCH_SIZE || idy === updates[appId].length - 1) {
-                    try {
-                        await countlyDb.collection(crashgroupCollection).bulkWrite(buffer, { ordered: false });
-                    }
-                    catch (err) {
-                        errCount += 1;
-                        console.error(`Failed updating collection ${crashgroupCollection}`, err);
-                    }
-                    finally {
-                        buffer.length = 0;
-                    }
+                });
+
+                if (requests.length === BATCH_SIZE) {
+                    countlyDb.collection(`app_crashgroups${app._id}`).bulkWrite(requests, function(err){
+                        if (err) {
+                            console.error(err);
+                        }
+                    });
+                    requests = [];
                 }
-            }
-        }
-
-        if (errCount === 0) {
-            try {
-                await countlyDb.collection(crashgroupCollection).updateOne(
-                    { _id: 'meta' },
-                    { $set: { app_version_list_added: true } },
-                );
-            }
-            catch (err) {
-                console.error(`Failed updating collection ${crashgroupCollection} meta`, err);
-            }
-
-            try {
-                await countlyDb.collection(crashgroupCollection).ensureIndex({ app_version_list: 1 }, { background: true });
-            }
-            catch (err) {
-                console.error(`Failed updating collection ${crashgroupCollection} index`, err);
-            }
-        }
-        else {
-            console.error(`${errCount} batches failed when updating collection ${crashgroupCollection}`);
+            }, function() {
+                if(requests.length > 0) {
+                    countlyDb.collection(`app_crashgroups${app._id}`).bulkWrite(requests, function(err){
+                        if (err) {
+                            console.error(err);
+                        }
+                        done();
+                    });
+                }
+                else {
+                    done();
+                }
+            });
         }
-    }
-
-    countlyDb.close();
-    console.log('Crashgroup data update done');
+        asyncjs.eachSeries(apps, update, function() {
```

Comment:
We dont need asyncjs,
we can just use native promises or async/await, better performance too